src/Metadata/Resource/Factory/UriTemplateResourceMetadataCollectionFactory.php

```diff
@@ -29,7 +31,7 @@ final class UriTemplateResourceMetadataCollectionFactory implements ResourceMeta
 {
     private $triggerLegacyFormatOnce = [];
 
-    public function __construct(private readonly LinkFactoryInterface $linkFactory, private readonly PathSegmentNameGeneratorInterface $pathSegmentNameGenerator, private readonly ?ResourceMetadataCollectionFactoryInterface $decorated = null)
+    public function __construct(private readonly LinkFactoryInterface $linkFactory, private readonly PathSegmentNameGeneratorInterface $pathSegmentNameGenerator, private readonly ?ResourceMetadataCollectionFactoryInterface $decorated = null, private readonly ?LoggerInterface $logger = new NullLogger())
```

Comment:
Shouldn't we inject the logger with a setter instead (see `LoggerAwareTrait`)? I think having a direct dependency is too much for a logger.

Reply:
we do like this everywhere and its totally optional I don't see the issue

Reply:
IMO the logger has no value as a direct dependency. It is totally optional and the class can be used without it without losing any benefit. I personally see it as a best practice.